### DISCUSSION TITLE
Devtools questions

### DISCUSSION THREAD

**Discussion by jonphipps** [Laravel-Backpack/community-forum#879] [Q&A (Help)] [answered]
1. Can Devtools start from an existing database or migrations? I can generate fresh migrations, but it doesn't look like they're of any use. I have about 30 existing tables.

2. Will Devtools generate code in any base path? I only manage users and roles/permissions in the admin path.

**Comment by jcastroa87** [comment, marked as answer]
Hi @jonphipps 

Devtools is created to help developers handle CRUD creation on a web interface; if you want to start from an existing db schema, it will not be so useful unless you have Model created too, because from there, it is possible to create the CRUD, it Factory and Seeders.

Devtools generate code in the admin path.

Cheers.

**Reply by jonphipps** [reply]
I doubt that it's going to be helpful.
